internal/bminventory/inventory.go

```diff
@@ -1925,6 +1930,28 @@ func (b *bareMetalInventory) updateDhcpNetworkParams(updates map[string]interfac
 	}
 	return nil
 }
+func (b *bareMetalInventory) updatePlatformParams(params installer.UpdateClusterParams, cluster *common.Cluster, updates map[string]interface{}) {
+	platform := params.ClusterUpdateParams.Platform
+	if platform == nil || platform.Type == "" {
```

Comment:
How can user unset a profile?

Reply:
You will always have a platform.  Platform is nil when it is not part of the `UpdateClusterParams`.
For replacing the cluster just patch `{ "platform": { "type": "baremetal" } }`

Reply:
please make sure that when you define vsphere then switch to baremetal that you don't get the old vscphere parameterse.